### PR TITLE
Fixes #21 Drag-and-drop isn't working with IE11

### LINKED ISSUE #21: Drag-and-drop isn't working with IE11 (and let's face it, probably other versions of IE)
When dropping a file from Windows Explorer into IE11, I get this error in the console: `SCRIPT87: Invalid argument.`.

It then proceeds to highlight [this line](https://github.com/tim-evans/ember-file-upload/blob/85b48afd2694f5e95fd6d8590b16910d09ef5e24/addon/components/file-dropzone/component.js#L121) from `file-dropzone/component.js`:

```javascript
let url = evt.dataTransfer.getData('text/uri-list');
```

I've scoured Google looking for a way to help fix this. I tried changing `text/uri-list` to `URL` [per MSDN](https://msdn.microsoft.com/en-us/library/ms536436(v=vs.85).aspx) with no success. (`console.log(url)` ends up evaluating to `undefined`.)

I'm not sure if you're interested in supporting IE or if this even works in Edge, but I thought that I would log what I've run into here if anyone was interested in fixing it.

### PR DESCRIPTION
**Pull request by chrisdpeters**
This silences the IE errors mentioned in #21.

I'm still learning the JavaScripts, so let me know if something could be written better or tested or something.

### CODE REVIEW

File: addon/components/file-dropzone/component.js [hunk 1]
```diff
@@ -118,16 +118,30 @@ export default Ember.Component.extend({
...
     // Testing support for dragging and dropping images
     // from other browser windows
-    let url = evt.dataTransfer.getData('text/uri-list');
-    let html = evt.dataTransfer.getData('text/html');
-    if (html) {
+    let supportsUrls, supportsHtml, html, url;
+
+    try {
+      html = evt.dataTransfer.getData('text/html');
+      supportsHtml = true;
```
Comment (chrisdpeters) [inline addon/components/file-dropzone/component.js]: @tim-evans In hindsight, I guess it doesn't even really make much sense to store values like `supportsHtml` and `supportsUrls` the way that I'm doing it here. The `if` tests below would still receive `undefined` `html` and `url` values.

I know in #21, you had mentioned storing `supportsHtml` and `supportsUrls` properties on the component (e.g., `this.set('supportsHtml, true)`), but I wasn't sure how you'd want to use those in context of the component.

Comment (tim-evans) [inline addon/components/file-dropzone/component.js]: This looks good; caching this globally may be nice 😄

Comment (tim-evans) [inline addon/components/file-dropzone/component.js]: (at the top of the file)

Comment (chrisdpeters) [inline addon/components/file-dropzone/component.js]: Would that even be possible if you need an event object to begin with? I'm learning a lot here, so I'd appreciate any insight that you have.

Comment (tim-evans) [inline addon/components/file-dropzone/component.js]: Sure! I understand where you're going with it, but I'm thinking something closer to moving the `let` to the top of the file and only checking the first time we get into `drop`.

```js
let supportsHtml = null;

// ... later
if (supportsHtml == null) {
  try {
    supportsHtml = true;
  } catch () {
    supportsHtml = false;
  }
}
```

Comment (chrisdpeters) [inline addon/components/file-dropzone/component.js]: Makes sense! I thought you were saying to cache the actual logic at the top of the file, which left me scratching my head. (I spend roughly 50% of my day scratching my head, so this isn't out of the norm.)

See new commit 48a826d5ae12509dc4dd3d6693d75646497e04bc, which I agree is great for performance.

**Pull request merged**